packages/fxa-auth-server/lib/routes/subscriptions/stripe.ts

```diff
@@ -158,38 +163,66 @@ export class StripeHandler {
         ...(pConfig.capabilities?.[ALL_RPS_CAPABILITIES_KEY] || [])
       );
 
-      const capabilityKeys = Object.keys(metadata).filter((key) =>
-        key.startsWith('capabilities:')
+      result = ClientIdCapabilityMap.merge(
+        result,
+        clientIdCapabilityMapFromMetadata(
+          plan.plan_metadata || {},
+          'capabilities:'
+        )
       );
-      for (const key of capabilityKeys) {
-        const clientId = key.split(':')[1];
-        const capabilities = commaSeparatedListToArray((metadata as any)[key]);
-        capabilitiesByClientId[clientId] = (
-          capabilitiesByClientId[clientId] || []
-        ).concat(capabilities);
-      }
+
+      result = ClientIdCapabilityMap.merge(
+        result,
+        clientIdCapabilityMapFromMetadata(
+          plan.product_metadata || {},
+          'capabilities:'
+        )
+      );
+
       if (pConfig.capabilities) {
```

Comment:
This is code from the ProductConfig implementation, and is no longer necessary. Could you please create a separate ticket to remove this code?

Reply:
👍 [FXA-8488](https://mozilla-hub.atlassian.net/browse/FXA-8488) has been created to remove ProductConfig implementation.